src/main/java/qtlog/FilesystemController/FileMonitor.java

```diff
@@ -26,16 +46,112 @@ public void notifyObserver() {
     public FileDTO getFileInformation() {
         return newestFile;
     }
-
+    
     @Override
-    public void setFolderpath(String folderpath) {
-        // TODO Auto-generated method stub
+    public void run() {
+        try {
+            this.WatchDir(this.filepath);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        this.processEvents();
     }
 
-    //example code, delete later
-    public void setNewestFile(FileDTO newFile){
-        this.newestFile = newFile;
-        this.notifyObserver();
+    /**
+     * Register the given directory, and all its sub-directories, with the
+     * WatchService.
+     */
+    private void registerAll(final Path start) throws IOException {
+        // register directory and sub-directories
+        Files.walkFileTree(start, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+                throws IOException
+            {
+                registerKey(dir);
+                System.out.format("Registering dir: %s ...\n", dir);
+                return FileVisitResult.CONTINUE;
+            }
+        });
+    }
+
+    /**
+     * Register the given directory with the WatchService
+     */
+    private void registerKey(Path dir) throws IOException  {
+        WatchKey key = dir.register(watchservice, StandardWatchEventKinds.ENTRY_MODIFY);
+        if (trace) {
+            Path prev = keys.get(key);
+            if (prev == null) {
+                System.out.format("register: %s\n", dir);
+            } else {
+                if (!dir.equals(prev)) {
+                    System.out.format("update: %s -> %s\n", prev, dir);
+                }
+            }
+        }
+        keys.put(key, dir);
+    }
+
+    /**
+     * Creates a WatchService and registers the given directory
+     */
+    private void WatchDir(Path dir) throws IOException {
+        this.watchservice = FileSystems.getDefault().newWatchService();
+        this.keys = new HashMap<WatchKey,Path>();
+
+        System.out.format("Scanning %s ...\n", dir);
+        this.registerAll(dir);
+        System.out.println("Done.");
+
+        // enable trace after initial registration
+        this.trace = true;
+    }
+
+    /**
+     * Process all events for keys queued to the watcher
+     */
+    void processEvents() {
+        WatchEvent<Path> lastEvent = null;
+        while(true) {
+
+            // wait for key to be signalled
+            WatchKey key;
+            try {
+                key = watchservice.take();
+            } catch (InterruptedException x) {
+                return;
+            }
+
+            Path dir = keys.get(key);
+            if (dir == null) {
+                System.err.println("WatchKey not recognized!!");
```

Comment:
If it detects a new directory inside the arc log folder (and it isn't registered), shouldn't we register it instead of just throwing an error? Else bosses the user hasn't got any prior logs of will be skipped until the program is restarted.

Reply:
Totally, will check it out